compute/compute_api.go

```diff
@@ -0,0 +1,5 @@
+package compute
```

Comment:
Will it be shared with clusters API as well?

Reply:
We're not sure yet, but I'd rather break it out sooner than later.

Reply:
please don't add a package just with one `ComputeSpec`. 

use Go SDK
https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.9.0/service/jobs

otherwise, move this type to `jobs/resource_job.go` for now